File: Solutions/001-Two-Sum/two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dictMap = {}
        for index, value in enumerate(nums):
            if target - value in dictMap:
                return dictMap[target - value], index
            else:
                dictMap[value] = index


s = Solution()
print("TestCase:")
print(s.twoSum([2, 7, 11, 15], 9))

File: Solutions/001-Two-Sum/two_sum.go
package main

func twoSum(nums []int, target int) []int {
	reverseMap := map[int]int{}
	for i := 0; i < len(nums); i++ {
		var index, ok = reverseMap[nums[i]]
		if ok && i != index {
			ans := []int{index, i}
			return ans
		}
		reverseMap[target-nums[i]] = i
	}
	return nil
}

func main() {
	nums := []int{3, 2, 4}
	target := 6
	var ans = twoSum(nums, target)
	for i, v := range ans {
		println(i, v)
	}
}
